Lib/pickle.py

```diff
@@ -731,6 +767,46 @@ def save_bytes(self, obj):
         self.memoize(obj)
     dispatch[bytes] = save_bytes
 
+    def save_bytearray(self, obj):
+        if self.proto < 5:
+            if not obj:  # bytearray is empty
+                self.save_reduce(bytearray, (), obj=obj)
+            else:
+                self.save_reduce(bytearray, (bytes(obj),), obj=obj)
+            return
+        n = len(obj)
+        if n >= self.framer._FRAME_SIZE_TARGET:
+            self._write_large_bytes(BYTEARRAY8 + pack("<Q", n), obj)
+        else:
+            self.write(BYTEARRAY8 + pack("<Q", n) + obj)
+    dispatch[bytearray] = save_bytearray
+
+    def save_picklebuffer(self, obj):
+        if self.proto < 5:
+            raise PicklingError("PickleBuffer can only pickled with "
+                                "protocol >= 5")
+        with obj.raw() as m:
+            if not m.contiguous:
+                raise PicklingError("PickleBuffer can not be pickled when "
+                                    "pointing to a non-contiguous buffer")
+            in_band = True
+            if self._buffer_callback is not None:
+                in_band = bool(self._buffer_callback(obj))
+            if in_band:
+                # Write data in-band
+                # XXX we could avoid a copy here
```

Comment:
Should we do something about this? This affects only the pure Python implementation right?

Reply:
Only the pure Python implementation indeed. I don't care much, but the XXX is there in case someone like @ogrisel wants to squeeze out more performance out of it :-)

Reply:
IRL discussion report with @ogrisel : As long as the C implementation is efficient, it is not that big of a deal if the python implementation is suboptimal.

Reply:
Cool, thank you :-)

Reply:
Please elaborate the comment. Maybe just:

```
# XXX we could avoid a copy here 
# The C implementation supports zero-copy
```